### PR TITLE
Ios integration

### PR DESCRIPTION
**Pull request by justshreyas**

### CODE REVIEW

File: ios/Classes/SplitDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        
+        result(flutterResult)
+        
+        
+    }
+    
+    public func getTreatments(splitNames: [String],attributes: [String:Any], result: @escaping FlutterResult){
+        
+        
+        let treatments = self.client?.getTreatments(splits: splitNames, attributes: attributes)
```
Comment (guptaShantanu) [inline ios/Classes/SplitDelegate.swift:60]: In case of null client, What will this method return?

Comment (justshreyas) [reply ios/Classes/SplitDelegate.swift:60]: I think null, can check in the example app by commenting out the initialization of the plugin, and directly calling the `getTreatments`

File: ios/Classes/SplitDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        
+        self.client?.on(event: SplitEvent.sdkReady){
+            result(nil)
+            print("!! Split init !!")
+        }
+        
+        self.client?.on(event: SplitEvent.sdkReadyTimedOut) {
+            result(nil)
+            print("!! Split SDK timed out !!")
+        }
```
Comment (justshreyas) [inline ios/Classes/SplitDelegate.swift:28]: Missing `onSdkReady`

File: ios/Classes/SplitDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        let config = SplitClientConfig()
+        
+        //Factory
+        let builder = DefaultSplitFactoryBuilder()
+        let factory = builder.setApiKey(apiKey).setKey(user).setConfig(config).build()
+        
+        //Client
+        client = factory?.client
+        
+        self.client?.on(event: SplitEvent.sdkReady){
```
Comment (justshreyas) [inline ios/Classes/SplitDelegate.swift:20]: @guptaShantanu

**Pull request merged**